swiftui/PasswordlessSwiftUI/PasswordlessSwiftUI/ContentView.swift

```diff
@@ -12,72 +19,83 @@ import Firebase
 /// Main view where user can login in using Email Link Authentication.
 struct ContentView: View {
   @State private var email: String = ""
-  @State private var presentSheet = false
+  @State private var isPresentingSheet = false
 
   var body: some View {
 
     NavigationView {
-      VStack(spacing: 50.0) {
+      VStack(alignment: .leading) {
         Text("Authenticate users with only their email, no password required!")
-          .frame(maxWidth: .infinity, alignment: .leading)
-        
-        Spacer()
-          .frame(height: 0.0)
-        
+          .padding(.bottom, 60)
+
         CustomStyledTextField(
-          text: $email, placeholder: "Email", symbol: "person.circle.fill"
+          text: $email, placeholder: "Email", symbolName: "person.circle.fill"
         )
 
         CustomStyledButton(title: "Send Sign In Link") {
-          sendSignInLink(to: email)
+          sendSignInLink()
         }
+        .disabled(email.isEmpty)
 
         Spacer()
       }
       .padding()
       .navigationBarTitle("Passwordless Login")
-
     }
     .onOpenURL { (url) in
       let link = url.absoluteString
       if Auth.auth().isSignIn(withEmailLink: link) {
-        passwordlessSignIn(email: email, link: link) { (success) in
-          self.presentSheet = success
+        passwordlessSignIn(email: email, link: link) { user, error in
+          isPresentingSheet = user?.isEmailVerified ?? false
         }
       }
     }
-    .sheet(isPresented: $presentSheet) {
+    .sheet(isPresented: $isPresentingSheet) {
       SuccessView(email: $email)
     }
+  }
+
+  // MARK: - Firebase 🔥
 
+  private func sendSignInLink() {
+    let actionCodeSettings = ActionCodeSettings()
+    actionCodeSettings.url = URL(string: "https://passwordlessswiftui.page.link/demo_login")
+    actionCodeSettings.handleCodeInApp = true
+    actionCodeSettings.setIOSBundleID(Bundle.main.bundleIdentifier!)
+
+    Auth.auth().sendSignInLink(toEmail: email, actionCodeSettings: actionCodeSettings)
   }
 
+  private func passwordlessSignIn(email: String, link: String,
+                                  completion: @escaping (_ user: User?, _ error: Error?) -> ()) {
```

Comment:
We should take advantage of the `Result` type here instead of providing two optionals in the callback.